week1/homework/src/server.js

```diff
@@ -10,6 +10,28 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        response.setHeader('Content-Type', 'application/json');
+        response.end(JSON.stringify({ state: state }));
+        break;
+      case '/add':
+        response.setHeader('Content-Type', 'application/json');
+        response.end(JSON.stringify({ state: (state += 1) }));
+        break;
+      case '/subtract':
+        response.setHeader('Content-Type', 'application/json');
+        response.end(JSON.stringify({ state: (state -= 1) }));
+        break;
+      case '/reset':
+        response.setHeader('Content-Type', 'application/json');
+        response.end(JSON.stringify({ state: (state = 10) }));
+        break;
+      default:
+        response.statusCode = 404;
+        response.setHeader('Content-Type', 'application/json');
+        response.end(JSON.stringify({ error: 'Not found' }));
+    }
```

Comment:
💡 PRO TIP: `response.end` at the very end can save you a lot of headaches in case someone forgets to call end it in some of the switch-cases.